modules/core/src/main/java/org/apache/synapse/mediators/store/MessageStoreMediator.java

```diff
@@ -74,9 +67,18 @@ public boolean mediate(MessageContext synCtx) {
 
 
         if(synCtx != null) {
-
+            MessageStore messageStore = synCtx.getConfiguration().getMessageStore(messageStoreName);
             if(messageStore != null) {
 
+                if (messageStore.getParameters().get(PRODUCER_GUARANTEED_DELIVERY) != null) {
```

Comment:
Better assign messageStore.getParameters().get(PRODUCER_GUARANTEED_DELIVERY) to a variable and reuse.